apps/emqx/test/emqx_common_test_helpers.erl

```diff
@@ -947,3 +961,24 @@ latency_up_proxy(off, Name, ProxyHost, ProxyPort) ->
         [],
         [{body_format, binary}]
     ).
+
+%%-------------------------------------------------------------------------------
+%% Testcase teardown utilities
+%%-------------------------------------------------------------------------------
+
+get_on_exit_callbacks() ->
+    persistent_term:get({?MODULE, on_exit}, []).
+
+put_on_exit_callbacks(Funs) ->
+    persistent_term:put({?MODULE, on_exit}, Funs).
+
+on_exit(Fun) ->
```

Comment:
To avoid concurrent tests interfering each other, the callbacks may need a better namespace.
e.g.
```suggestion
on_exit(ID, Fun) ->
```

`-define(ON_EXIT(Fun), on_exit({?MODULE, ?FUNCTION_NAME}, Fun)).`
and in `end_per_testcase`, should use `{?MODULE, Case}` as the key.